src/Kmd.Logic.CitizenDocuments.Client/CitizenDocumentsClient.cs

```diff
@@ -89,6 +94,144 @@ public async Task<CitizenDocumentUploadResponse> UploadAttachmentWithHttpMessage
+
+            if (uploadResponse.StatusCode != System.Net.HttpStatusCode.OK)
+            {
+                throw new Exception("Upload Failed");
+            }
+
+            citizenDocumentUploadRequestModel.Id = documentId;
+            citizenDocumentUploadRequestModel.DocumentUrl = $"{containerAddress.ToString()}/{storageDocName}";
+            citizenDocumentUploadRequestModel.Status = "Completed";
+
+            var updateResponse = await client.UpdateDataToDbWithHttpMessagesAsync(
+                                 subscriptionId: new Guid(this.options.SubscriptionId),
+                                 request: citizenDocumentUploadRequestModel).ConfigureAwait(false);
+            switch (updateResponse.Response.StatusCode)
+            {
+                case System.Net.HttpStatusCode.OK:
+                    return (CitizenDocumentUploadResponse)updateResponse.Body;
+
+                case System.Net.HttpStatusCode.Unauthorized:
+                    throw new CitizenDocumentsException("Unauthorized", updateResponse.Body as string);
+
+                default:
+                    throw new CitizenDocumentsException("Invalid configuration provided to access Citizen Document service", updateResponse.Body as string);
+            }
+        }
+
+        private static async Task<UploadResponseModel> UploadDocumentToAzureStorage(Stream document, string documentName, CloudBlobContainer container, int size = 5000000)
+        {
+            if (documentName == null)
+            {
+                throw new ArgumentNullException(nameof(documentName));
+            }
+
+            CloudBlockBlob blob = container.GetBlockBlobReference(documentName);
+            try
+            {
+                int bytesRead;
+                int blockNumber = 0;
+                List<string> blockList = new List<string>();
+                do
+                {
+                    Console.WriteLine(blockNumber);
+                    blockNumber++;
+                    string blockId = $"{blockNumber:0000000}";
+                    string base64BlockId = Convert.ToBase64String(Encoding.UTF8.GetBytes(blockId));
+                    byte[] buffer = new byte[size];
+                    bytesRead = await document.ReadAsync(buffer, 0, size).ConfigureAwait(false);
+                    await blob.PutBlockAsync(base64BlockId, new MemoryStream(buffer, 0, bytesRead), null).ConfigureAwait(false);
+                    blockList.Add(base64BlockId);
+                }
+                while (bytesRead == size);
+                await blob.PutBlockListAsync(blockList).ConfigureAwait(false);
+                document.Dispose();
+                return new UploadResponseModel()
+                {
+                    StatusCode = System.Net.HttpStatusCode.OK,
+                    Message = "Upload Successfull",
+                };
+            }
+            catch (ArgumentNullException ex)
```

Comment:
In what case this exception happens? `ArgumentNullException` can be usually handled by just checking if value is null.

Reply:
Changed